pwa-devdocs/src/technologies/basic-concepts/internationalization/index.md

```diff
@@ -0,0 +1,150 @@
+
+The first and second entry uses the original `en_US` locale text as the ID for the translated text.
+This is the same approach used in Magento.
+
+The third and fourth entry uses an alternative, dot notation approach for the defining the ID for translated text.
+This helps to identify the component and element where the text appears.
+
+Both approach have their pros and cons, and developers are free to choose which approach works for them when they develop their own components and storefronts.
+
+### Message syntax
+
+The i18n framework accepts the same [message syntax][] as the underlying `react-intl` library.
+Along with static text, this syntax supports variables, dates, and even conditional formatting.
+
+To translate text with variables, pass in a mapping object to the `values` prop in the `FormattedMessage` component.
+
+{% raw %}
+
+```jsx
+return(
+    <FormattedMessage
+    id="Hi, {name}"
+    defaultMessage="Hi, {name}"
+    values={{
+        name: 'Veronica',
+    }}
+    />
+)
+```
+
+{% endraw %}
+
+When using the `formatMessage()` function, pass in the mapping object as the second parameter.
+
+```jsx
+const text = formatMessage(
+    { id:"Hi, {name}", defaultMessage="Hi, {name}"},
+    { name: currentUser.firstname }
+);
+```
+
+For more details, see the [message syntax][] documentation at FormatJS.
+
+## Language packages and plugins
+
+Language packages provide translation data for one or more locales.
+They are also used to override the text in the same locale.
+
+Unlike the core Magento application, which install language packages through Composer, PWA Studio storefronts install language packages as NPM dependencies.
+
+An NPM dependency is a language package if it meets the following criteria:
+
+- The package contains an intercept file
+- The intercept file sets the special feature `i18n` flag to `true` for the package
+- The package contains an `i18n` directory
+- The `i18n` directory contains a dictionary file with a locale formatted name
+
+<!-- TODO: Create an in-depth tutorial for creating a language package extension -->
+
+## Build process
+
+To optimize runtime performance, the i18n framework compiles all the translation data during the build process.
+
+The following is a high level summary of the actions the i18n framework takes to compile the translation data:
+
+1. Scans the `i18n` directory in the project and installed language packages for `.json` files that have a locale formatted file name
+2. Generates an object with locales as keys that map to an array of files matching that locale
+3. Merges the files in the arrays to create a single dictionary object for a locale
+4. Creates a virtual module from this object that exposes a `__fetchLocaleData__` function
+5. Generates a dynamic import in the application for the virtual module
+
+During runtime, the `LocaleProvider` component uses the `__fetchLocaleData__` function to get the correct translation data for the current locale.
```

Comment:
I want to say the locale/store switcher component uses a graphql query to only allow switching using "enabled" locales. If that's true, it might be worth mentioning that even if the translation pack, `i18n/xx_yy.json` is provided, you still have to enable it in the backend to be able to utilize it in the UI.

cc @eug123 @davemacaulay

Reply:
Note added